v2 NeedAuthorization

Zasa-san opened the pull request:
fixes issue when trying to migrate and test v2 translation component on the client, since it uses NeedAuthorization which is connected to the store. This new component uses atoms.

Also, included in this PR, there is a new testing strategy for unit testing components with atoms.

v1 NeedAuthorization is left unchanged to avoid messy breaking changes.

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

Zasa-san committed:
v2 NeedAuthorization

Zasa-san committed:
Merge branch 'development' into migrate-needAuthorization

Zasa-san committed:
Merge branch 'development' into migrate-needAuthorization

mfacar reviewed (approved):
LGTM! Good catch 👏 

Just a minor comment regarding the location of app/react/V2/atoms/TestAtomStoreProvider.tsx that I'm leaving to your criteria; I think it could be better to have this file as part of a test folder and not between the production code.

Zasa-san committed:
apply the new testing method for components with atoms

Zasa-san committed:
move testing helper to testing folder

Zasa-san committed:
move test atom provider to testing folder

Zasa-san committed:
Merge branch 'migrate-needAuthorization' of https://github.com/huridocs/uwazi into migrate-needAuthorization

Zasa-san commented:
> LGTM! Good catch 👏
> 
> Just a minor comment regarding the location of app/react/V2/atoms/TestAtomStoreProvider.tsx that I'm leaving to your criteria; I think it could be better to have this file as part of a test folder and not between the production code.

Good idea! I have moved all v2 testing utils into a folder, and updated some older testing methods we used with atoms

Zasa-san committed:
Merge branch 'development' into migrate-needAuthorization

Zasa-san committed:
update imports

Zasa-san committed:
renamed file

Zasa-san committed:
Merge branch 'development' into migrate-needAuthorization

Zasa-san committed:
Merge branch 'development' into migrate-needAuthorization

mfacar reviewed (approved):
LGTM! 👏
